spec/factories/course_assessment_categories.rb

```diff
@@ -2,7 +2,7 @@
   sequence(:course_assessment_category_weight) { |n| n }
   sequence(:course_assessment_category_title) { |n| "Category #{n}" }
   factory :course_assessment_category, class: Course::Assessment::Category do
-    course
+    course { create(:course) }
```

Comment:
it doesn't have root_folder until saved: https://github.com/Coursemology/coursemology2/blob/master/app/models/course/material/folder.rb#L20